src/query/storage/fanout/storage.go

```diff
@@ -332,9 +333,42 @@ func (s *fanoutStorage) CompleteTags(
 
 	built := accumulatedTags.Build()
 	built.Metadata = metadata
+	built = applyOptions(built, options)
 	return &built, nil
 }
 
+func applyOptions(
```

Comment:
Hm, so for query results the handlers filters it out but for complete tags storage filters it out?

Perhaps this should just be done at the handler level?

Reply:
Think we discussed this offline; it's largely based on where parsing happens for the fetch itself; the DAG needs to be built before we can add it to the fetch path unfortunately